lib/galaxy/model/__init__.py

```diff
@@ -2630,7 +2630,9 @@ class PostJobAction(Base, RepresentById):
     workflow_step_id: Mapped[Optional[int]] = mapped_column(ForeignKey("workflow_step.id"), index=True)
     action_type: Mapped[str] = mapped_column(String(255))
     output_name: Mapped[Optional[str]] = mapped_column(String(255))
-    _action_arguments: Mapped[Optional[bytes]] = mapped_column(MutableJSONType)
+    _action_arguments: Mapped[Optional[Union[bool, Dict[str, Any]]]] = mapped_column(
```

Comment:
While we can have `bool` in the database the type checker should complain if you attempt this. Do things fail with
```suggestion
    _action_arguments: Mapped[Optional[Dict[str, Any]]] = mapped_column(
```
?

Reply:
It doesn't fail, but at line 2650 we are checking as if it's potentially a bool:

```python
if ... and self._action_arguments is True:
```

so I suppose a future version of mypy may complain about this.
I think this is correct.

Reply:
Then in that future version we ignore the type (or migrate the buggy values in the column). We should never ever set booleans in that column, any attempt to do so is a bug.